uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -143,23 +140,25 @@ define([
        * @return {Promise} empty promise if all required params are not set otherwise promise from server
        */
       vm.calculateBalanceChange = function () {
-        setDateAndTypes();
+        var that = this;
 
-        if (!canCalculateChange()) {
+        setDateAndTypes.call(that);
+
+        if (!canCalculateChange.call(that)) {
           return $q.resolve();
         }
 
         vm.error = null;
-        vm.loading.calculateBalanceChange = true;
-        return LeaveRequest.calculateBalanceChange(getParamsForBalanceChange())
+        that.loading.calculateBalanceChange = true;
+        return LeaveRequest.calculateBalanceChange(getParamsForBalanceChange.call(that))
```

Comment:
i did not want to mix 2 styles in code and hence used that-->this at places consistently